docs/user_authentication/third_party_token_management.md

```diff
@@ -0,0 +1,25 @@
+
+```sequence {theme: 'simple'}
+Title: Third party idp token management
+Pbench-Dashboard->Pbench-Server:[1] GET login page
+Pbench-Server->Pbench-Dashboard:[2] 200 login page
+Pbench-Dashboard->Pbench-Dashboard:[3] Generate Authorization request
+Pbench-Dashboard->Identity-Provider:[4] POST Authorization request
+Identity-Provider->Identity-Provider:[5] 200 Authenticate User
```

Comment:
No, the response for the line 7 i.e. `Pbench-Dashboard->Identity-Provider: POST Authorization request` is `Identity-Provider->Pbench-Dashboard: 200 Authorization code`. The authentication of the user only happens on Identity-provider's site.
So the following sequence happens:
1. Pbench-dashboard post the authorization request
2. Identity-provider authenticates the user and gets consent from the user for the Pbench client
3. Once the user agrees to the consent, Identity-provider sends a 200 response with `authorization_code` This code is a uuid-like number that is active for like a minute and can be used only once to retrieve the tokens on behalf of a user.
4. Pbench-dashboard takes the `authorization_code` and Post it back to the Identity-provider which then sends a 200 response with access and refresh token.

Does that make sense?

Reply:
Nikhil, I think your description there is missing a few steps and some magic.

When the authorization sequence starts, the Pbench Dashboard page is loaded in the Browser, and it's running the Dashboard Javascript.  Somehow, the Identity-provider page is loaded into the Browser so that the user can communicate directly with the Identity-provider, without the Dashboard as an intermediary.  I assume that this happens because the Dashboard requests it, but I'm unclear on what causes the Browser to load the page from the Identity-provider instead of routing the response back to the Dashboard.

The next part of the dialog is all under the control of the Identity-provider.  It requests the credentials from the User, validates them, and then requests consent from the user to share information with the Pbench Dashboard.  If that all goes as expected, then somehow the Identity-provider sends a response back to the Dashboard with the `authorization_code`, but I don't understand how this happens if the Identity-provider has been loading pages into the Browser.  (I imagine that there is some sort of "from" reference on the original authorization request from the Pbench Dashboard, but I don't know how the Identity-provider causes the Dashboard code to be reactivated so that it can receive the response.)

But, clearly it does happen, and the Dashboard resumes running, receives the response from the Identity-provider, and then uses the `authorization_code` to request the token-pair...and then we're off to the races.

So, I think you're missing a couple of Browser state transitions and a couple of bits of dialog.